From 5df391b9e6554b9ad4c1a1ffae85126744bdc56b Mon Sep 17 00:00:00 2001
From: Mark Fasheh <mfasheh@suse.de>
Date: Thu, 8 Sep 2011 17:40:01 -0700
Subject: [PATCH 20/20] btrfs: Remove BUG_ON from __finish_chunk_alloc()
Patch-mainline: 3.2?

btrfs_alloc_chunk() unconditionally BUGs on any error returned from
__finish_chunk_alloc() so there's no need for two BUG_ON lines. Remove the
one from __finish_chunk_alloc().

Signed-off-by: Mark Fasheh <mfasheh@suse.de>
---
 fs/btrfs/volumes.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/fs/btrfs/volumes.c
+++ b/fs/btrfs/volumes.c
@@ -3384,7 +3384,8 @@ static int __finish_chunk_alloc(struct b
 		device = map->stripes[index].dev;
 		device->bytes_used += stripe_size;
 		ret = btrfs_update_device(trans, device);
-		BUG_ON(ret);
+		if (ret)
+			goto out_free;
 		index++;
 	}
 
@@ -3431,6 +3432,7 @@ static int __finish_chunk_alloc(struct b
 					     item_size);
 	}
 
+out_free:
 	kfree(chunk);
 	return ret;
 }
